### PR TITLE
[Feature-10486][Python] Python-gate should support the use of full-name definitions for resources.

### LINKED ISSUE #10486: [Feature][Python] Python-gate should support the use of full-name definitions for resources.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Now when I define a Python task that requires importing resources.
I need to check the id of the resource in the database first.
And then I write code that looks something like this.
```    
    test_task = Python(
        name="测试任务",
        resource_list=[{"id": 64}, {"id": 65}, {"id": 66}, {"id": 67}, {"id": 68}],
        code=base_info_py_code
    )
```
resource_list should be allowed to be defined using the resource full-name .

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hiSandog**
Now when I define a Python task that requires importing resources via python-gate.
I need to check the id of the resource in the database first.
And then I write code that looks something like this.
```
    test_task = Python(
        name="测试任务",
        resource_list=[{"id": 64}, {"id": 65}, {"id": 66}, {"id": 67}, {"id": 68}],
        code=base_info_py_code
    )
```
resource_list should be allowed to be defined using the resource full-name ,like this

```
    test_task = Python(
        name="测试任务",
        resource_list=[{"resourceName": "/abce/aa.py"}, {"resourceName": "/abce/bb.py"}],
        code=base_info_py_code
    )
```

close: #10486

### CODE REVIEW

File: dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/process/ProcessServiceTest.java [hunk 1]
```diff
@@ -30,21 +30,7 @@
...
-import org.apache.dolphinscheduler.dao.entity.ProcessDefinitionLog;
-import org.apache.dolphinscheduler.dao.entity.ProcessInstance;
-import org.apache.dolphinscheduler.dao.entity.ProcessInstanceMap;
-import org.apache.dolphinscheduler.dao.entity.ProcessTaskRelationLog;
-import org.apache.dolphinscheduler.dao.entity.Resource;
-import org.apache.dolphinscheduler.dao.entity.TaskDefinitionLog;
-import org.apache.dolphinscheduler.dao.entity.TaskGroupQueue;
-import org.apache.dolphinscheduler.dao.entity.TaskInstance;
-import org.apache.dolphinscheduler.dao.entity.User;
+import org.apache.dolphinscheduler.dao.entity.*;
```
Comment (SbloodyS) [inline dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/process/ProcessServiceTest.java]: Please avoid import *

**Comment by zhongjiajie** [comment]
instead of using 
```py    
test_task = Python(
        name="测试任务",
        resource_list=[{"resourceName": "/abce/aa.py"}, {"resourceName": "/abce/bb.py"}],
        code=base_info_py_code
    )
```
I think maybe use 

```py
test_task = Python(
        name="测试任务",
        resource_list=[{"resourceName": "/abce/aa.py"}, {"resourceName": "/abce/bb.py"}],
        code=base_info_py_code
    )
```
is better

**Comment by hiSandog** [comment]
> instead of using
> 
> ```python
> test_task = Python(
>         name="测试任务",
>         resource_list=[{"resourceName": "/abce/aa.py"}, {"resourceName": "/abce/bb.py"}],
>         code=base_info_py_code
>     )
> ```
> 
> I think maybe use
> 
> ```python
> test_task = Python(
>         name="测试任务",
>         resource_list=[{"resourceName": "/abce/aa.py"}, {"resourceName": "/abce/bb.py"}],
>         code=base_info_py_code
>     )
> ```
> 
> is better

These two pieces of code look the same.I'm a little confused about what you mean @zhongjiajie

**Comment by zhongjiajie** [comment]
🤣 sorry my bad, I am sleepy last night and do not notice that. I mean the init proposal is support using path in Python API. so maybe we should only change some code in python gateway or add a new interface entrypoint to do that.
and for Python API side, we can simple use 

```py
test_task = Python(
        name="test",
        resource_list=["/abce/aa.py", "/abce/bb.py"],
        code=base_info_py_code
    )
```
and then add covert it to dict and add key `resourceName` in our python cod, you can see https://github.com/apache/dolphinscheduler/blob/4be3b877c3c143e2b5c58c017b641fd927c0dff5/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py#L201-L212 as example

**Comment by hiSandog** [comment]
> 🤣 sorry my bad, I am sleepy last night and do not notice that. I mean the init proposal is support using path in Python API. so maybe we should only change some code in python gateway or add a new interface entrypoint to do that. and for Python API side, we can simple use
> 
> ```python
> test_task = Python(
>         name="test",
>         resource_list=["/abce/aa.py", "/abce/bb.py"],
>         code=base_info_py_code
>     )
> ```
> 
> and then add covert it to dict and add key `resourceName` in our python cod, you can see
> 
> https://github.com/apache/dolphinscheduler/blob/4be3b877c3c143e2b5c58c017b641fd927c0dff5/dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py#L201-L212
> 
> as example

I know what you mean.I will close this pr and then submit  another one . @zhongjiajie

**Pull request closed without merging**